From 936badbeec5b1075194fb70cba2b963e7a53cc8f Mon Sep 17 00:00:00 2001
From: Stanislaw Gruszka <sgruszka@redhat.com>
Date: Fri, 5 Sep 2014 11:23:40 -0400
Subject: [kernel] PTR_RET is now PTR_ERR_OR_ZERO

Message-id: <1409916232-5882-4-git-send-email-sgruszka@redhat.com>
Patchwork-id: 91433
O-Subject: [RHEL7 PATCH 03/15] PTR_RET is now PTR_ERR_OR_ZERO
Bugzilla: 1136324
RH-Acked-by: John Linville <linville@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: John Greene <jogreene@redhat.com>

BZ#1136324

Backport of:

commit 6e8b8726ad503214ba66e34aed69aff41de33489
Author: Rusty Russell <rusty@rustcorp.com.au>
Date:   Mon Jul 15 11:19:32 2013 +0930

    PTR_RET is now PTR_ERR_OR_ZERO

    True, it's often used in return statements, but after much bikeshedding
    it's probably better to have an explicit name.

    (I tried just putting the IS_ERR check inside PTR_ERR itself and gcc
    usually generated no more code.  But that clashes current expectations
    of how PTR_ERR behaves, so having a separate function is better).

    Suggested-by: Julia Lawall <julia.lawall@lip6.fr>
    Suggested-by: "Michael S. Tsirkin" <mst@redhat.com>
    Cc: Julia Lawall <julia.lawall@lip6.fr>
    Cc: "Michael S. Tsirkin" <mst@redhat.com>
    Signed-off-by: Rusty Russell <rusty@rustcorp.com.au>

Brew:
https://brewweb.devel.redhat.com/taskinfo?taskID=7918052

Testing:
Tested on Intel, Atheros, Ralink wireless hardware.

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/include/linux/err.h b/include/linux/err.h
index f2edce2..ee365c4 100644
--- a/include/linux/err.h
+++ b/include/linux/err.h
@@ -52,7 +52,7 @@ static inline void * __must_check ERR_CAST(const void *ptr)
  return (void *) ptr;
 }
 
-static inline int __must_check PTR_RET(const void *ptr)
+static inline int __must_check PTR_ERR_OR_ZERO(const void *ptr)
 {
  if (IS_ERR(ptr))
   return PTR_ERR(ptr);
@@ -60,6 +60,9 @@ static inline int __must_check PTR_RET(const void *ptr)
   return 0;
 }
 
+/* Deprecated */
+#define PTR_RET(p) PTR_ERR_OR_ZERO(p)
+
 #endif
 
 #endif /* _LINUX_ERR_H */
-- 
1.7.1